src/user/user.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable, BadRequestException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { UserDto } from './dtos/user.dto';
+import { User } from './user.entity';
+import { UpdateUserDto } from './dtos/updateUser.dto';
+import { Organization } from 'src/organization/organization.entity';
+import { UpdateResult } from 'typeorm';
+import { Roles } from './roles.enumt';
+import * as bcrypt from 'bcrypt';
+
+@Injectable()
+export class UserService {
+    constructor(
+        @InjectRepository(User) private repo: Repository<User>
+    ) { }
+
+    async createUser(dto: UserDto): Promise<User> {
+        const hashedPass = await bcrypt.hash(dto.password, 10);
```

Comment:
@BeforeInsert @BeforUpdate = this kind of typeorm decorators can be used in you db model to bcrypt or do any manipulation with the data before inserting, so in this case you will not need to hash the password in service.